src/base/os.lua

```diff
@@ -80,6 +80,9 @@
 			if os.is("macosx") then
 				formats = { "lib%s.dylib", "%s.dylib" }
 				path = os.getenv("DYLD_LIBRARY_PATH") or ""
+			elseif os.is("cygwin") then
+				formats = { "lib%s.a", "%s.a", "lib%s.dll.a", "%s.dll.a" }
+				path = os.getenv("LD_LIBRARY_PATH") or ""

```

Comment:
Hmmm. Yeah, I'm semi-convinced.
It raises the question; what about mingw?
And what about support for cygwin cross-compiling under linux? I think that's equally important. cygwin/mingw is often the target when cross-compiling for windows.
I've had some similar thoughts wrt mingw recently. @starkos?